src/profile/notification-settings/PreferenceCheckbox.tsx

```diff
@@ -0,0 +1,65 @@
+import _ from 'lodash/fp';
+import React from 'react';
+import { Checkbox } from 'src/components/common';
+import { withErrorReporting } from 'src/libs/error';
+import { EventWorkspaceAttributes } from 'src/libs/events';
+import * as Utils from 'src/libs/utils';
+import {
+  notificationEnabled,
+  NotificationType,
+  updateNotificationPreferences,
+  updateUserAttributes,
+} from 'src/profile/notification-settings/utils';
+
+export interface NotificationCheckboxProps {
+  notificationKeys: string[];
+  notificationType: NotificationType;
+  label: string;
+  setSaving: (boolean) => void;
+  prefsData: Record<string, string>;
+  workspace: EventWorkspaceAttributes | undefined;
+}
+
+export const NotificationCheckbox = (props: NotificationCheckboxProps) => {
+  const { label, notificationKeys, notificationType, prefsData, setSaving, workspace } = props;
+  const checkboxProps = {
+    'aria-label': label,
+    checked: notificationEnabled(notificationKeys, prefsData),
+    onChange: _.flow(
+      Utils.withBusyState(setSaving),
+      withErrorReporting('Error saving notification preferences')
+    )(async (v) => {
+      await updateNotificationPreferences(notificationKeys, v, notificationType, workspace);
+    }),
+  };
+
+  return <Checkbox {...checkboxProps} />;
+};
+
+export interface UserAttributesCheckboxProps {
+  notificationKeys: string[];
+  notificationType: NotificationType;
+  label: string;
+  value: boolean;
+  disabled: boolean;
+  setSaving: (boolean) => void;
+}
+
+export const UserAttributesCheckbox = (props: UserAttributesCheckboxProps) => {
```

Comment:
Taken from https://github.com/DataBiosphere/terra-ui/blob/366a30ae3ae2b263355581e63a45de9eebc8017c/src/profile/notification-settings/NotificationSettings.js#L82, converted to TSX, and Ajax methods moved to utils for easier unit testing.